### PR TITLE
Use a seperate collection to store user's current pms.

### PR DESCRIPTION
**Pull request by kaushikgandhi**
According to #18 we had issues with user pms . Here is a patch for the issue .

### CODE REVIEW

File: waartaa/client/chat.js [hunk 1]
```diff
@@ -380,8 +380,9 @@ Handlebars.registerHelper('pms', function (id) {
   var server = UserServers.findOne({_id: id});
   var user = Meteor.user();
   var pms = [];
+   var userpms=UserPms.findOne({user_id: user._id});
```
Comment (rtnpro) [inline waartaa/client/chat.js]: space around `=`

File: waartaa/client/chat.js [hunk 2]
```diff
@@ -537,19 +538,24 @@ Template.user_menu.events = {
...
-    if (!profile)
-      profile = {connections: {}};
-    if (!profile.connections[server_id])
-      profile.connections[server_id] = {pms: {}};
-    if (!profile.connections[server_id].pms)
-      profile.connections[server_id].pms = {};
-    profile.connections[server_id].pms[nick] = '';
-    console.log(profile);
-    Meteor.users.update({_id: user._id}, {$set: {profile: profile}});
+    var userpms=UserPms.findOne({user_id: user._id});
```
Comment (rtnpro) [inline waartaa/client/chat.js]: same here

File: waartaa/client/chat.js [hunk 3]
```diff
@@ -537,19 +538,24 @@ Template.user_menu.events = {
...
+
+    console.log('updating pms');
+    // if (!profile)
+    //   profile = {connections: {}};
+    if (!userpms)
+      userpms = {pms: {}};
+    if (!userpms.pms)
+      userpms.pms = {};
+    userpms.pms[nick] = '';
+    //console.log(profile);
```
Comment (rtnpro) [inline waartaa/client/chat.js]: Remove this line.

File: waartaa/client/chat.js [hunk 4]
```diff
@@ -537,19 +538,24 @@ Template.user_menu.events = {
...
+    UserPms.upsert({user_id: user._id, user_server_id: server_id,user_server_name: server_name,user: user.username}, {$set: {pms: userpms.pms}});
```
Comment (rtnpro) [inline waartaa/client/chat.js]: space after `,`

File: waartaa/client/chat.js (hunk 3 above)
Comment (kaushikgandhi) [inline waartaa/client/chat.js]: that line was written by you i just commented it :)

File: waartaa/client/chat.js [hunk 5]
```diff
@@ -651,9 +657,13 @@ Template.server_pm_menu.events = {
...
     var nick = $target.data('user-nick');
     var profile = user.profile;
-    var pms = user.profile.connections[user_server_id].pms;
+    var userpms=UserPms.findOne({user_id: user._id});
+    var pms = userpms.pms;
+     var server = UserServers.findOne({_id: user_server_id});
+    var server_name = server.name;
     delete pms[nick];
-    Meteor.users.update({_id: user._id}, {$set: {profile: profile}});
+    UserPms.upsert({user_id: user._id, user_server_id: user_server_id,user_server_name: server_name,user: user.username}, {$set: {pms: pms}});
```
Comment (rtnpro) [inline waartaa/client/chat.js]: Space after `,`

File: waartaa/server/lib/02_irc.js [hunk 6]
```diff
@@ -384,9 +384,12 @@ IRCHandler = function (user, user_server) {
...
             if (to == client.nick) {
                 var profile = user.profile;
-                profile.connections[user_server._id].pms[nick] = "";
+                
+                
                 Fiber(function () {
-                    Meteor.users.update({_id: user._id}, {$set: {profile: profile}});
+                    var userpms=UserPms.findOne({user_id: user._id}) || {pms: {}};
+                    userpms.pms[nick] = "";
+                    UserPms.upsert({user_id: user._id, user_server_id: user_server._id,user_server_name: user_server.name,user: user.username}, {$set: {pms: userpms.pms}});
```
Comment (rtnpro) [inline waartaa/server/lib/02_irc.js]: Space after `,`

File: waartaa/server/publish.js [hunk 7]
```diff
@@ -76,6 +76,9 @@ Meteor.publish(
     return cursor;
 });
 
+Meteor.publish('user_pms', function () {
+  return UserPms.find({user_id: this.userId});
+});
 /*
```
Comment (rtnpro) [inline waartaa/server/publish.js:82]: Add a newline before this

**Comment by rtnpro** [comment]
@kaushikgandhi Thanks for the fix :+1: Merging it.

**Pull request merged**